Implement Response Models

PineappleIOnic opened the pull request:
This PR will implement response models into the Rust SDK

TorstenDittmann commented:
How are generics being handlded?

Meaning, when I use `listDocuments`, can I tell the method to type the items of `documents` to my type?

PineappleIOnic commented:
Generics are handled by Serde using the Value Enum which enables a user to parse them how they like using the Serde Library https://docs.serde.rs/serde_json/value/enum.Value.html

For instance the document struct looks like so:
```rust
pub struct Document {
    #[serde(rename(serialize = "id", deserialize = "$id"))]
    pub id: EmptyOption<String>,
    #[serde(rename(serialize = "collection", deserialize = "$collection"))]
    pub collection: EmptyOption<String>,
    #[serde(rename(serialize = "permissions", deserialize = "$permissions"))]
    pub permissions: EmptyOption<Permissions>,
    pub data: HashMap<String, Value>, // <-- This handles generic data
}
```

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Improving Go Templates.

* Fixing #8
* Adding Support for go modules
* Using JSON Payload for Documents

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
Adding Test and bug fixes in Client Template

phaus committed:
activating go tests

phaus committed:
activating go tests

phaus committed:
activating go tests

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
we don't need that one here

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
trying to fix the test build

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
updating test case

phaus committed:
updating test case

phaus committed:
updating test case

phaus committed:
updating test case

phaus committed:
updating test case

phaus committed:
updating test case

phaus committed:
activating go tests

phaus committed:
activating go tests

phaus committed:
activating go tests

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

PineappleIOnic commented:
Removing `rust-language-support` in favour of `rust-response-models`. I'm going to close the branch down and rename `rust-response-models` to be the main branch as there is no reason to merge it into there.

phaus committed:
Creating a first testable version for go 1.12

phaus committed:
Creating a first testable version for go 1.12

phaus committed:
Creating a first testable version for go 1.12

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
fixing test cases

phaus committed:
making go tests run

phaus committed:
making go tests run

phaus committed:
making go tests run

phaus committed:
test updates

phaus committed:
test updates

phaus committed:
test updates

phaus committed:
upstream rebase and test config fix

phaus committed:
upstream rebase and test config fix

phaus committed:
upstream rebase and test config fix

phaus committed:
support api errors

phaus committed:
support api errors

phaus committed:
support api errors

phaus committed:
file uploadt WIP

phaus committed:
file uploadt WIP

phaus committed:
file uploadt WIP

phaus committed:
fixing upload test

phaus committed:
fixing upload test

phaus committed:
fixing upload test

phaus committed:
adding download test, improving client response

phaus committed:
adding download test, improving client response

phaus committed:
adding download test, improving client response

phaus committed:
activating all languages

phaus committed:
activating all languages

phaus committed:
activating all languages

phaus committed:
git ignore cleanup

phaus committed:
git ignore cleanup

phaus committed:
git ignore cleanup

phaus committed:
some updates regarding PR comments

phaus committed:
some updates regarding PR comments

phaus committed:
some updates regarding PR comments

phaus committed:
some update in regards of review feedbacks

phaus committed:
some update in regards of review feedbacks

phaus committed:
some update in regards of review feedbacks

phaus committed:
updating binary file download handling

phaus committed:
updating binary file download handling

phaus committed:
activating swift client and server tests again

phaus committed:
activating swift client and server tests again

phaus committed:
activating swift client and server tests again

phaus committed:
fixing testrun

phaus committed:
fixing testrun

phaus committed:
fixing testrun

phaus committed:
fixes based on feedback

phaus committed:
fixes based on feedback

phaus committed:
fixes based on feedback

phaus committed:
updating main example

phaus committed:
updating main example

phaus committed:
updating main example

phaus committed:
updating client code in regards of review feedback

phaus committed:
updating client code in regards of review feedback

phaus committed:
updates reflecting feedback from review

phaus committed:
updates reflecting feedback from review

phaus committed:
updates reflecting feedback from review

phaus committed:
updates reflecting feedback from review

phaus committed:
updates reflecting feedback from review

phaus committed:
removing static appwrite from sources

phaus committed:
removing static appwrite from sources

phaus committed:
updating client template

phaus committed:
updating client template

phaus committed:
adding last fixes for rebase with upstream

phaus committed:
adding last fixes for rebase with upstream

phaus committed:
adding last fixes for rebase with upstream

phaus committed:
trying to fix the other tests

phaus committed:
trying to fix the other tests

phaus committed:
file cleanup

phaus committed:
file cleanup

phaus committed:
file cleanup

phaus committed:
ignore phpunit cache

phaus committed:
fixing test cases for with dummy repo name

phaus committed:
fixing test cases for with dummy repo name

phaus committed:
using struct as a client response

phaus committed:
using struct as a client response

phaus committed:
using struct as a client response

phaus committed:
updating generated documentation

phaus committed:
updating generated documentation

phaus committed:
ignoring .phpunit.result.cache

phaus committed:
fixing go test case

abnegate committed:
Fix swift server expected responses

lohanidamodar committed:
Merge pull request #322 from appwrite/feat-cookie-tests

abnegate committed:
Merge remote-tracking branch 'origin/master' into feat-swift-async-await

# Conflicts:
#	composer.lock
#	templates/swift/Sources/Client.swift.twig
#	tests/languages/swift-client/Tests.swift

abnegate committed:
Merge remote-tracking branch 'origin/master' into feat-parallel-env-test

# Conflicts:
#	composer.lock

abnegate committed:
Merge remote-tracking branch 'origin/master' into 0.14.x

abnegate committed:
Merge remote-tracking branch 'origin/0.14.x' into feat-resumable-upload-ruby

abnegate committed:
Merge remote-tracking branch 'origin/0.14.x' into feat-resumable-upload-swift

abnegate committed:
Merge remote-tracking branch 'origin/master' into feat-resumable-upload-kotlin

abnegate committed:
Merge remote-tracking branch 'origin/master' into feat-resumable-upload-android

abnegate committed:
Fix random access read

abnegate committed:
Fix random access read

abnegate committed:
Fail on get existing file with ID error

abnegate committed:
Fix optional interpolation

abnegate committed:
Fix symbol use

abnegate committed:
Merge remote-tracking branch 'origin/0.14.x' into feat-resumable-upload-kotlin

abnegate committed:
Merge remote-tracking branch 'origin/0.14.x' into feat-resumable-upload-android

abnegate committed:
Fix cookies using async/await API

abnegate committed:
Rebase fix-android-cookies on remote-tracking brach 'origin/master-base'

abnegate committed:
Merge pull request #415 from appwrite/feat-resumable-upload-swift

Feat resumable upload swift

abnegate committed:
Merge pull request #414 from appwrite/feat-resumable-upload-ruby

Feat resumable upload ruby

abnegate committed:
Merge pull request #412 from appwrite/feat-resumable-upload-android

Feat resumable upload android

abnegate committed:
Merge pull request #411 from appwrite/feat-resumable-upload-kotlin

Feat resumable upload kotlin

abnegate committed:
Merge pull request #416 from appwrite/feat-parallel-env-test

Parallel env CI tests

abnegate committed:
Merge remote-tracking branch 'origin/master' into fix-android-12-cookies

abnegate committed:
Merge remote-tracking branch 'origin/master' into feat-swift-async-await

abnegate committed:
Merge remote-tracking branch 'origin/master' into 0.14.x

# Conflicts:
#	tests/KotlinJava11Test.php

abnegate committed:
Fix cookie package ref

lohanidamodar committed:
Merge pull request #408 from appwrite/fix-android-12-cookies

Add fixed cookie dependency for Android 12+

abnegate committed:
Merge pull request #409 from appwrite/feat-swift-async-await

Feat swift async await

phaus committed:
migrating tests to new setup

phaus committed:
cleanup

phaus committed:
cleanup

phaus committed:
naming is difficult

phaus committed:
re-adding large file response

phaus committed:
adding another missing response type

phaus committed:
deactivate failing test

Meldiron committed:
Added JSON parsing

christyjacob4 committed:
Merge pull request #403 from tanimdiucse123/patch-1

Grammatical mistake

christyjacob4 committed:
Merge pull request #419 from appwrite/fix-cli-user-prefs

Fix: JSON object parsing

christyjacob4 committed:
Merge pull request #410 from PineappleIOnic/fix-delete-archive-on-fail

Fix Archive not being deleted when the deploy request fails

phaus committed:
updates regarding review

lohanidamodar committed:
upgrade package info plus

lohanidamodar committed:
beta test

eldadfux committed:
Merge pull request #421 from appwrite/feat-latest-dependencies-dart

Flutter upgrade dependencies

lohanidamodar committed:
Merge branch 'master' into go-templates-update

abnegate committed:
Merge remote-tracking branch 'origin/master' into 0.14.x

# Conflicts:
#	templates/cli/lib/commands/command.js.twig
#	templates/swift/Sources/Client.swift.twig
#	templates/swift/Sources/Services/Service.swift.twig

abnegate committed:
Merge fixes

lohanidamodar committed:
rollback test spec host to appwrite.io

abnegate committed:
Revert hardcoded header names

lohanidamodar committed:
Merge pull request #405 from appwrite/0.14.x

Meldiron committed:
Fix install command (powershell)

lohanidamodar committed:
Merge pull request #353 from phaus/go-templates-update

lohanidamodar committed:
file examples apart from image

Meldiron committed:
Fix example CLI version

lohanidamodar committed:
update path

lohanidamodar committed:
Merge pull request #426 from appwrite/feat-flutter-examples

christyjacob4 committed:
Merge pull request #423 from appwrite/fix-windows-cli-command

Fix: Powershell install

PineappleIOnic committed:
Start adding attributes

PineappleIOnic committed:
Merge remote-tracking branch 'upstream/master' into rust-response-models

PineappleIOnic committed:
Get Rust SDK buildable again

PineappleIOnic committed:
First Fully functional SDK Build for Rust

+ Fixed using .unwrap on non-required objects causing a Panic
- Fixed SDK Attempting to convert no content responses into JSON
- Fixed a few display impl with objects
- Fixed the way we deal with enum structs
- Got attributes working correctly
